am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -89,7 +90,14 @@ public void grantExplicitResourceAccess(@NotNull @Valid ExplicitAccessGrant acce
                         .attribute(attributePermission.getKey())
                         .relationship(accessGrant.getRelationship())
                         .build())
-                    .forEach(dao::createAccessManagementRecord));
+                    .forEach(expAccessRecord -> {
+                        if (nonNull(accessGrant.getRelationship())) {
+                            dao.createAccessManagementRecord(expAccessRecord);
+                        } else {
```

Comment:
can you change the name of the methods as below -
1. grantAccessManagementWithNullRelationship()
2. grantAccessManagementWithNotNullRelationship()

Reply:
done